src/utils/DropdownButton.vue

```diff
@@ -0,0 +1,160 @@
+        >
+            <span class="visually-hidden">Toggle Dropdown</span>
+        </button>
+        <ul class="dropdown-menu" :class="{ show: expanded }" :aria-labelledby="uniqueHtmlId">
+            <li v-for="item in items" :key="item.value">
+                <a
+                    class="dropdown-item"
+                    :class="{ active: currentValue === item.value }"
+                    :data-cy="`dropdown-item-${item.title.toLowerCase()}`"
+                    @click="selectItem(item)"
+                >
+                    {{ $t(item.title) }}
+                </a>
+            </li>
+        </ul>
+    </div>
+</template>
+
+<script>
+import { Dropdown } from 'bootstrap'
+import { randomIntBetween } from '@/utils/numberUtils'
+
+/**
+ * Represents an option in the select made for a dropdown. If no value is given, the title of the
+ * item will be considered the value.
+ *
+ * All given title (for main button or items) will go through the i18n services before being
+ * rendered (it's not mandatory to directly pass a translated string, the translation key is sufficient)
+ */
+export class DropdownItem {
+    constructor(title, value = null) {
+        this._title = title
+        this._value = value || title
+    }
+
+    get title() {
+        return this._title
+    }
+    get value() {
+        return this._value
+    }
+}
+
+/**
+ * Manages a Bootstrap dropdown.
+ *
+ * All items must be passed as instances of DropdownItem (imported from the same file as the
+ * component). The active item (the one with the .active CSS class) will be defined by comparing the
+ * props `currentValue` with each item's value.
+ *
+ * It is possible to have the dropdown attached to an extra button with a caret, instead of all
+ * inline. For that you have to add the with-toggle-button HTML attribute to the component use
+ */
+export default {
+    props: {
+        title: {
+            type: String,
+            required: true,
+        },
+        currentValue: {
+            type: Object,
+            required: true,
+        },
+        items: {
+            type: Array,
+            required: true,
+            validator: (items) => {
+                // checking that we received an array of only DropdownItem instances with at least one item
+                if (Array.isArray(items) && items.length > 1) {
+                    return (
+                        items.filter((item) => item instanceof DropdownItem).length === items.length
+                    )
+                }
+                return false
+            },
+        },
+        withToggleButton: {
+            type: Boolean,
+            default: false,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    emits: ['click', 'select:item'],
+    data() {
+        return {
+            // generating a unique HTML ID for this dropdown
+            uniqueHtmlId: `dropdown-${randomIntBetween(0, 10000)}`,
```

Comment:
Not sure here that `randomIntBetween(0, 10000)` is safe, if the application grows and that you have many drop downs items you might have collisions ! Question does the ID needs to be unique within the dropdown button or within the whole application ?
Maybe something like vue-uid (https://dev.to/jshimkoski/ssr-friendly-unique-html-ids-in-vue-3-1og9) plugin might be a good idea ? Or using a more robust uuid like `uuid` or `nanoid` npm packages.

Reply:
I was wondering if I should add `uuid` lib or not, and thought that we shouldn't have more than 3 or 4 dropdown at the same time on the screen, and so that a simple random number should be good enough instead of adding yet another lib to the stack already present in `package.json`

Reply:
Ok I see your point

Reply:
there might be a possibility to use `getUid()` from OpenLayers, but I also didn't want to add OL import in part of the app that doesn't really use mapping...